### PR TITLE
Magnum: Cluster Templates: Create, Get, Delete, List (#1003)

### PR DESCRIPTION
**Pull request by openstacker**
Prior to starting a PR, please make sure you have read our
[contributor tutorial](https://github.com/gophercloud/gophercloud/tree/master/docs/contributor-tutorial).

Prior to a PR being reviewed, there needs to be a Github issue that the PR
addresses. Replace the brackets and text below with that issue number.

For #[PUT ISSUE NUMBER HERE]

Links to the line numbers/files in the OpenStack source code that support the
code in this PR:

[PUT URLS HERE]

### CODE REVIEW

**Comment by jtopjian** [comment]
@openstacker Thank you for beginning the work on this. We'll need to split this up into 4 PRs though: one for each of create, get, delete, and list. The contributor tutorial has references to other PRs to use as an example.

In addition, there needs to be links to the Python service API code which can confirm the implementation is correct. This is outlined in the "Code Hunting" section of the contributor tutorial.

Please let me know if you have any questions.

**Comment by openstacker** [comment]
@jtopjian Thanks for the comments and I'm aware of the policy submitting one PR per API action. But I saw this commit https://github.com/gophercloud/gophercloud/commit/1ad7a96d69a2e8105920431ee428843b6d1ee4c3 so I'm wondering if the policy has been changed. 

I'm happy to split this into 4 PRs.

**Comment by jtopjian** [comment]
@openstacker Yes, please split into individual PRs - especially when beginning with contributions to Gophercloud. After a while we can discuss submitting multiple features at once - but that would only be after prior discussion

The commit you referenced is from this PR: https://github.com/gophercloud/gophercloud/pull/1059

Which I mention:

> I'm going to break protocol and do a bulk merge of this since due to the time constraints of core reviewers (including myself), getting around to reviewing individual PRs would take months.

It was certainly a one-off occurrence which I don't plan to do regularly. :)

**Comment by openstacker** [comment]
@jtopjian, thanks for the clarification. All good. Cheers.

File: script/stackenv [hunk 1]
```diff
@@ -11,8 +11,11 @@ _EXTGW_ID=$(openstack network show public -c id -f value)
 _IMAGE=$(openstack image list | grep -i cirros | head -n 1)
 _IMAGE_ID=$(echo $_IMAGE | awk -F\| '{print $2}' | tr -d ' ')
 _IMAGE_NAME=$(echo $_IMAGE | awk -F\| '{print $3}' | tr -d ' ')
+_COE_IMAGE=$(openstack image list | grep -i Fedora-Atomic | head -n 1)
```
[the remark this answers is not in the document]
Comment (openstacker) [reply script/stackenv]: Awesome, thank you for all the comments. Will address them soon. Cheers.

File: acceptance/openstack/containerinfra/v1/clustertemplate_test.go [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+		COE:               "kubernetes",
+		ImageID:           choices.COEImageID,
+		KeypairID:         "default",
+		DockerVolumeSize:  5,
+		FlavorID:          choices.FlavorID,
+		MasterFlavorID:    choices.FlavorID,
+		ExternalNetworkID: choices.ExternalNetworkID,
+		NetworkDriver:     "calico",
+	}
+	ct, err := clustertemplates.Create(client, opts).Extract()
```
[the remark this answers is not in the document]
Comment (openstacker) [reply acceptance/openstack/containerinfra/v1/clustertemplate_test.go]: That's a very good point. Cheers.

File: script/acceptancetest [hunk 3]
```diff
@@ -46,6 +46,12 @@ if [[ $? != 0 ]]; then
   failed=1
 fi
 
+# Container Infra v1
+go test -v -timeout $timeout -tags "fixtures acceptance" ./acceptance/openstack/containerinfra/v1/
```
[the remark this answers is not in the document]
Comment (openstacker) [reply script/acceptancetest]: ACK.

File: acceptance/openstack/containerinfra/v1/clustertemplate_test.go [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+	// Create a new cluster template
+	opts := clustertemplates.CreateOpts{Name: "k8s",
+		COE:               "kubernetes",
+		ImageID:           choices.COEImageID,
+		KeypairID:         "default",
+		DockerVolumeSize:  5,
+		FlavorID:          choices.FlavorID,
+		MasterFlavorID:    choices.FlavorID,
+		ExternalNetworkID: choices.ExternalNetworkID,
+		NetworkDriver:     "calico",
```
[the remark this answers is not in the document]
Comment (openstacker) [reply acceptance/openstack/containerinfra/v1/clustertemplate_test.go]: I will drop it to use the default 'flannel' driver.

File: script/stackenv (hunk 1 above)
[the remark this answers is not in the document]
Comment (jtopjian) [reply script/stackenv]: Good point. Let's leave this as-is for now and we'll see if an issue arises when this is integrated with OpenLab.

**Pull request closed without merging**